mslice/plotting/plot_window/plot_figure.py

```diff
@@ -65,74 +73,50 @@ def _get_min(self, data, absolute_minimum=-np.inf):
-                mappable.colorbar.remove()
-                if vmin == float(0):
-                    vmin = 0.001
-                norm = colors.LogNorm(vmin, vmax)
-                mappable.set_norm(norm)
-                self.canvas.figure.colorbar(mappable)
-            elif not plot_config.logarithmic and type(mappable.norm) != colors.Normalize:
-                mappable.colorbar.remove()
-                norm = colors.Normalize(vmin, vmax)
-                mappable.set_norm(norm)
-                self.canvas.figure.colorbar(mappable)
+        if xy_config['x_log']:
+            xdata = [ll.get_xdata() for ll in current_axis.get_lines()]
+            xmin = self.get_min(xdata, absolute_minimum=0.)
+            current_axis.set_xscale('symlog', linthreshx=pow(10, np.floor(np.log10(xmin))))
+            if self.get_min(xdata) > 0:
+                xy_config['x_range'] = (xmin, xy_config['x_range'][1])
         else:
-            if plot_config.xlog:
-                xdata = [ll.get_xdata() for ll in current_axis.get_lines()]
-                xmin = self._get_min(xdata, absolute_minimum=0.)
-                current_axis.set_xscale('symlog', linthreshx=pow(10, np.floor(np.log10(xmin))))
-                if self._get_min(xdata) > 0:
-                    plot_config.x_range = (xmin, plot_config.x_range[1])
-            else:
-                current_axis.set_xscale('linear')
-            if plot_config.ylog:
-                ydata = [ll.get_ydata() for ll in current_axis.get_lines()]
-                ymin = self._get_min(ydata, absolute_minimum=0.)
-                current_axis.set_yscale('symlog', linthreshy=pow(10, np.floor(np.log10(ymin))))
-                if self._get_min(ydata) > 0:
-                    plot_config.y_range = (ymin, plot_config.y_range[1])
-            else:
-                current_axis.set_yscale('linear')
-
-        current_axis.set_xlim(*plot_config.x_range)
-        current_axis.set_ylim(*plot_config.y_range)
-
-        legend_config = plot_config.legend
-        for handle in legend_config.handles:
-            handle.set_label(legend_config.get_legend_text(handle))
-
-        # To show/hide errorbars we will just set the alpha to 0
-        if plot_config.errorbar is not None:
-            self._set_errorbars_shown_state(plot_config.errorbar)
-
-        # The legend must be set after hiding/showing the error bars so the errorbars on the legend are in sync with
-        # the plot (in terms of having/not having errorbars)
-        if legend_config.visible:
-            leg = current_axis.legend()
-            leg.draggable()
+            current_axis.set_xscale('linear')
+        if xy_config['y_log']:
+            ydata = [ll.get_ydata() for ll in current_axis.get_lines()]
+            ymin = self.get_min(ydata, absolute_minimum=0.)
+            current_axis.set_yscale('symlog', linthreshy=pow(10, np.floor(np.log10(ymin))))
+            if self.get_min(ydata) > 0:
```

Comment:
Is there a reason to do this calculation again rather than just using the value calculated 2 lines above?